lib/compilers/rga.ts

```diff
@@ -0,0 +1,205 @@
+// POSSIBILITY OF SUCH DAMAGE.
+
+import path from 'path';
+
+import {readdir, rename, writeFile} from 'fs-extra';
+
+import {ExecutionOptions} from '../../types/compilation/compilation.interfaces';
+import {ParseFilters} from '../../types/features/filters.interfaces';
+import {BaseCompiler} from '../base-compiler';
+import * as exec from '../exec';
+import {logger} from '../logger';
+
+interface ASICSelection {
+    asic?: string;
+    error?: string;
+    printASICs?: boolean;
+}
+
+// RGA := Radeon GPU Analyzer
+export class RGACompiler extends BaseCompiler {
+    private dxcPath: string;
+
+    static get key() {
+        return 'rga';
+    }
+
+    constructor(info: any, env: any) {
+        super(info, env);
+
+        this.compiler.supportsIntel = false;
+        this.dxcPath = this.compilerProps(`compiler.${this.compiler.id}.dxcPath`);
+        logger.debug(`RGA compiler ${this.compiler.id} configured to use DXC at ${this.dxcPath}`);
+    }
+
+    override optionsForFilter(filters: ParseFilters, outputFilename: any, userOptions?: any): any[] {
+        return [outputFilename];
+    }
+
+    extractASIC(dxcArgs: string[]): ASICSelection {
+        // Scan dxc args for an `--asic` argument that should be stripped and passed later to RGA
+        // Default to RDNA2
+        let asic = 'gfx1030';
+        let printASICs = true;
+        for (let i = 0; i !== dxcArgs.length; ++i) {
+            const arg = dxcArgs[i];
+            if (arg === '--asic') {
+                // NOTE: the last arguments are the input source file and -spirv, so check
+                // if --asic immediately precedes that
+                if (i === dxcArgs.length - 3) {
+                    return {
+                        error: '--asic flag supplied without subsequent ASIC!',
+                    };
+                }
+                asic = dxcArgs[i + 1];
+                // Do a quick sanity check to determine if a valid ASIC was supplied
+                if (!asic.startsWith('gfx')) {
+                    return {
+                        // eslint-disable-next-line max-len
+                        error: `The argument immediately following --asic doesn't appear to be a valid ASIC. Please supply an ASIC from the following options:`,
+                        printASICs: true,
+                    };
+                }
+                // Remove these two arguments from the dxcArgs list
+                dxcArgs.splice(i, 2);
+
+                // If the user supplied a specific ASIC, don't bother printing available ASIC options
+                printASICs = false;
+                break;
+            }
+        }
+
+        return {
+            asic,
+            printASICs,
+        };
+    }
+
+    execTime(startTime: bigint, endTime: bigint): string {
+        return ((endTime - startTime) / BigInt(1000000)).toString();
+    }
+
+    override async exec(filepath: string, args: string[], execOptions: ExecutionOptions): Promise<any> {
```

Comment:
I think it's better to override `runCompiler()`, as the `exec()` function is also used to request `--version` at startup and then this implementation would be very confusing